File: decompiled/sources/com/hjq/base/action/ResourcesAction.java
package com.hjq.base.action;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.drawable.Drawable;
import androidx.core.content.ContextCompat;

public interface ResourcesAction {
    int getColor(int i);

    Context getContext();

    Drawable getDrawable(int i);

    Resources getResources();

    String getString(int i);

    String getString(int i, Object... objArr);

    <S> S getSystemService(Class<S> cls);

    /* renamed from: com.hjq.base.action.ResourcesAction$-CC  reason: invalid class name */
    public final /* synthetic */ class CC {
        public static Resources $default$getResources(ResourcesAction _this) {
            return _this.getContext().getResources();
        }

        public static String $default$getString(ResourcesAction _this, int i) {
            return _this.getContext().getString(i);
        }

        public static String $default$getString(ResourcesAction _this, int i, Object... objArr) {
            return _this.getResources().getString(i, objArr);
        }

        public static Drawable $default$getDrawable(ResourcesAction _this, int i) {
            return ContextCompat.getDrawable(_this.getContext(), i);
        }

        public static int $default$getColor(ResourcesAction _this, int i) {
            return ContextCompat.getColor(_this.getContext(), i);
        }

        public static <S> Object $default$getSystemService(ResourcesAction _this, Class cls) {
            return ContextCompat.getSystemService(_this.getContext(), cls);
        }
    }
}

File: decompiled/sources/com/hjq/base/action/ResourcesAction.kt
package com.hjq.base.action

import android.content.Context
import android.content.res.Resources
import android.graphics.drawable.Drawable
import androidx.core.content.ContextCompat

interface ResourcesAction {
    fun getContext(): Context

    fun getResources(): Resources {
        return getContext().resources
    }

    fun getString(id: Int): String {
        return getContext().getString(id)
    }

    fun getString(id: Int, vararg args: Any): String {
        return getResources().getString(id, *args)
    }

    fun getDrawable(id: Int): Drawable? {
        return ContextCompat.getDrawable(getContext(), id)
    }

    fun getColor(id: Int): Int {
        return ContextCompat.getColor(getContext(), id)
    }

    fun <S> getSystemService(serviceClass: Class<S>): S? {
        return ContextCompat.getSystemService(getContext(), serviceClass)
    }

    object CC {
        @JvmStatic
        fun `$default$getResources`(resourcesAction: ResourcesAction): Resources {
            return resourcesAction.getContext().resources
        }

        @JvmStatic
        fun `$default$getString`(resourcesAction: ResourcesAction, id: Int): String {
            return resourcesAction.getContext().getString(id)
        }

        @JvmStatic
        fun `$default$getString`(resourcesAction: ResourcesAction, id: Int, vararg args: Any): String {
            return resourcesAction.getResources().getString(id, *args)
        }

        @JvmStatic
        fun `$default$getDrawable`(resourcesAction: ResourcesAction, id: Int): Drawable? {
            return ContextCompat.getDrawable(resourcesAction.getContext(), id)
        }

        @JvmStatic
        fun `$default$getColor`(resourcesAction: ResourcesAction, id: Int): Int {
            return ContextCompat.getColor(resourcesAction.getContext(), id)
        }

        @JvmStatic
        fun <S> `$default$getSystemService`(
            resourcesAction: ResourcesAction,
            serviceClass: Class<S>
        ): S? {
            return ContextCompat.getSystemService(resourcesAction.getContext(), serviceClass)
        }
    }
}
